app/views.py

```diff
@@ -238,6 +108,33 @@ def events_list():
     return results, status.HTTP_200_OK
 
 
+@app.route("/api/events/search/", methods=['GET', 'POST'])
+def filter_or_search_events():
+    """Search or Filter the events list"""
+    page = request.args.get('page', 1, type=int)
+    category = request.args.get('category')
+    location = request.args.get('location')
+    if category and location:
+        filterd = Events.query.filter_by(category=category, location=location)
+    elif category:
+        filterd = Events.query.filter_by(category=category)
+    else:
+        filterd = Events.query.filter_by(location=location)
+
```

Comment:
good implementation of the filter. what would happen if the user gave neither the location nor the category? let's say they passed in the description?